buildkite/build_project.sh

```diff
@@ -206,10 +210,16 @@ case "$REPO_FULL_NAME" in
         ;;
 
     ldc-developers/ldc)
-        git submodule update --init
+        cd runtime && git clone --depth 5 https://github.com/ldc-developers/druntime
+        git clone --depth 5 https://github.com/ldc-developers/phobos
```

Comment:
>If you push more than five commits to the LDC's druntime or Phobos fork, this will fail.

No idea what this means, they have a lot more than five commits. What this does is simply pull the last five commits, has nothing to do with pushing.

>AFAICT it's possible to tell git to do shallow clones

Read you own SO link from before, which I saw before you pointed me to it: it doesn't work that well. It kinda, sorta works if you're only checking out HEAD or a tag or something- I lost track- but people finally resort to what I did here, cloning each submodule you want separately and manually.

>waste those 2 seconds

Time is not the issue, it's bandwidth, as I've said before in this thread. Shallow cloning saves something like 97% of the bandwidth needed for the submodules, which adds up across thousands of CI runs, which is presumably why you shallow clone all these other CI repos too.

If this setup ends up being flaky, we can always stabilize it later.

Reply:
> No idea what this means, they have a lot more than five commits. What this does is simply pull the last five commits, has nothing to do with pushing.

But I think it could happen that LDC master uses different checked out sub repositories?
If not, that's great, but then `--depth 1` would be even better.

Reply:
No, the ldc forks of druntime and phobos each have an `ldc` branch, which is the only branch from which ldc `master` checks out, and is the main branch that will be checked out for each submodule here. `--depth 1` may not work if someone commits to one of the submodule `ldc` branches and ldc `master` tries to check out an older submodule commit before the submodule hash is updated, so 5 tries to account for that. See above discussion.